src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const KeyAndValues = sourceString.split(';');
+  const end = {};
+
+  KeyAndValues.forEach((element) => {
+    const preElement = element.split(':');
+
+    if (preElement[0]) {
+      preElement[0] = preElement[0].trim();
+    }
+
+    if (preElement[1]) {
+      preElement[1] = preElement[1].trim();
+    }
```

Comment:
The if conditions to check the existence of 'preElement[0]' and 'preElement[1]' are not necessary if you trim the values unconditionally. You can directly trim the values without the if statements, as calling trim on an undefined value would cause an error, which is not handled here.